### PR TITLE
[TESTED, BUT NOT WORKING QUITE RIGHT YET] An alternative to removing cloning

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

This PR is an alternative PR to the infamous cloning removal PR (https://github.com/tgstation/tgstation/pull/48668).

This PR adds the cloning memory loss feature that some other servers have to /tg/station. People who are cloned will now receive two messages (one when they start being cloned, and one when their clone emerges from their cloning pod) that will tell them that their character does NOT remember the cause, circumstances, or details of their death.

Note that the other methods of reviving people that exist (replica pods, surgery, defibrillation, etc.) will NOT cause cloning memory loss, which is an incentive for people to try those methods before resorting to using the cloner.

## Why It's Good For The Game

This PR elegantly solves the main problem with the cloning machine: Why use the other, more creative (for now; just wait until someone posts a step by step process for how to do the headswapping technique in the medical doctor guide on the wiki)/interactive means of reviving someone when the magic one button revive machine exists? Well, if you're too lazy (or new or incompetent) to use something other than the cloner, your victim won't remember the name of who murdered them, what type of antag their murderer was, if their death even WAS a murder, etc.

At the same time, this PR _doesn't_ screw over medbay/the station when a murderboner, a deathvirus, or a blob causes half a station's worth of corpses to show up on Medbay's doorstep. If something has wiped out half of the station, you likely already know what it is and how it's killing everyone, so you don't need to spend 2-5 minutes headswapping and then reviving each and every corpse; you can offload some of the work to the cloner instead (while still working on reviving a few of the corpses via other means to speed up the process).

And who knows, maybe this cloning memory loss will deescalate a conflict or two between non-antags. A guy can hope.

And yes, I am aware that there will likely be a rough transition week or so in which people will scream out the names of their murderers before reading their (bigtext) cloning memory loss messages. But hey, I'll take a week or so of that instead of the shitshow that a multi-week testmerge of the full cloning removal PR will probably cause.

**tldr;
When there are too many corpses for a doctor to skillfully revive, they can clone some of them to make things easier on themselves. When there's just a slow trickle of corpses, a doctor can pursue more esoteric means of revival so that their patients can tell them who killed them, and their patients might even _thank_ their doctors for _not _ cloning them. I mean, it's not _likely_, but it _could_ happen...**

**I would also like to formally request for this PR to NOT be linked to on Reddit until AFTER it is either merged, testmerged, or closed, since if that happens, the maintainers will likely dismiss any approval or positive comments that this PR has as coming from "mindless Redditors who don't even play on /tg/". You can get your 100 upvotes _after_ oranges closes this PR just because of its title.**

## To-Do List/Possible Additional Features to Add in Future PRs?
- Make this PR actually work properly (currently, only the second message is being sent: https://github.com/tgstation/tgstation/pull/48700).
- Add cloning bans that keep people from being cloned but allow them to be revived via other means (preferably with visible signs/messages to doctors that inform them of this), for admins to give to people who just can't seem to get it through their thick skulls that cloning memory loss is now a thing.
- Decrease the effectiveness of upgraded stock parts for cloning pods? I'd only go for this if the addition of cloning memory loss isn't enough to placate oranges/the cloning-removal crowd.
- Add a sheet of diamond to the list of components needed to construct each cloning pod, and possibly also increase how long it takes to clone someone? Again, I'd only do this if doing it is necessary to placate the cloning-removal crowd, but kriskog mentioned that he didn't like the idea of just making it take longer for a cloning pod to clone someone because he feared that people would just construct more cloning pods. Adding a rare material to the list of things needed to construct a cloning pod could serve as a deterrent against constructing an extra 5 of them near roundstart (and it'd make diamonds a bit more useful).

## Changelog
:cl: ATHATH
balance: CLONED PEOPLE NOW NO LONGER REMEMBER WHAT KILLED THEM. People revived via non-cloning means do not suffer from this cloning memory loss; consider using non-cloning methods of revival if Medbay isn't being swamped with corpses.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
🤢

**Comment by mrhugo13** [comment]
Question though, shouldn't this be ran by headmins because it's something the admins would have to enforce?

**Comment by ATH1909** [comment]
> Question though, shouldn't this be ran by headmins because it's something the admins would have to enforce?

It might have to be, but so would a complete removal of cloning, I think.

And hey, at least this PR'll make more people aware that cloning memory loss is an _option_ than a random thread in the ideas subforum (or reply to the cloning megathread there) that only 2 people will reply to would.

**Comment by trollbreeder** [comment]
Travis man bad

**Comment by EdgeLordExe** [comment]
the problem with cloning isnt the rp element. The cloning is removed because it requires no skill nor any thought to revive someone making death meaningless, Every other method of revival either requires some kind of work being done beforehand , some kind of maintanance or active resuscitation. Cloning is literally as braindead as it can be , just drop off then click a button and leave

**Comment by mrhugo13** [comment]
> 
> 
> > Question though, shouldn't this be ran by headmins because it's something the admins would have to enforce?
> 
> It might have to be, but so would a complete removal of cloning, I think.
> 
> And hey, at least this PR'll make more people aware that cloning memory loss is an _option_ than a random thread in the ideas subforum (or reply to the cloning megathread there) that only 2 people will reply to would.

Headmins probably don't need balance changes run by them, given that their job is completely separate from that. Things that they _will_ have to enforce through rules definetly would though.

**Comment by pireamaineach** [comment]
This fails to actually solve the issue. Cloning devalues life as a whole, the problem is how people act before they're cloned because of it, not how they act after they're cloned.

**Comment by improvedname** [comment]
ITS NOT LIKE REPLICA PODS EXIST JUST SAYING CLONE REMOVAL HATERS

**Comment by ATH1909** [comment]
https://cdn.discordapp.com/attachments/326826573951008769/664887557372379186/cloning_issue.PNG

Hm, the first message isn't being sent to the clonee for some reason; only the second one.

Any idea(s) why?

**Comment by ATH1909** [comment]
> the problem with cloning isnt the rp element. The cloning is removed because it requires no skill nor any thought to revive someone making death meaningless, Every other method of revival either requires some kind of work being done beforehand , some kind of maintanance or active resuscitation. Cloning is literally as braindead as it can be , just drop off then click a button and leave

Yes, because we have to assume that our medical doctors will be new or braindead. Not everyone is a master surgeon, and unlike with most other roles, a medical doctor not knowing what they're doing (without the existence of cloning) will cause people to remain dead for the rest of a 60-90 minute round (which is the round length that oranges is trying to push us towards). Do you know what kind of STRESS people new medical doctors will be put under when they have ten corpses at their door, people are screaming at them and calling them an incompetent chucklefuck who deserves to die in a fire, and they're being told to do something called "headswapping" (or to use defibs, which are notoriously hard for newer players to figure out how to use) when they barely know how to swap their own fucking hands? You need 0 hours of playtime in this game to play Medical Doctor; do not expect medical doctors to know how to do _anything_ harder than pushing button that make dead man come back (even our current medical doctors sometimes struggle with cloning people).

Our population of good doctors is already critically low; we don't need to steepen the learning curve of medbay even _further_ and drive even more people away from trying out medical doctor with yet more complexity. Cloning should be an "oh god, I don't know what to do" button that lets your patient get back into the round if you're a newer/inexperienced/overwhelmed medical doctor who _shockingly_ doesn't know how to do literal fucking brain surgery. Note that it _also_ shouldn't be a first resort button that people will yell at you for NOT using, but my PR addresses that problem by adding an inherent downside to cloning (you don't know who killed you).

**Comment by pireamaineach** [comment]
Could it not be true that the reason for most doctors being new is because the skill ceiling for these jobs are critically low ?

**Comment by ATH1909** [comment]
Also, complex != fun. Once you _do_ learn how to headswap, asking for monkeys from genetics and/or xenobio every single damn round, making a mutadone smoke grenade for them every damn round, and actually performing the headswapping surgeries every single damn time a corpse whose heart has decayed to uselessness is brought into Medbay is going to become very tedious, very fast.

**Comment by ATH1909** [comment]
> Could it not be true that the reason for most doctors being new is because the skill ceiling for these jobs are critically low ?

Yes, because there's no reason to NOT just clone a corpse.

This PR adds a (good) reason to revive corpses via non-cloning means without making doing that a _necessity_. More experienced doctors can use the more advanced techniques, while less experienced ones can still fall back on cloning when shit hits the fan.

**Comment by ATH1909** [comment]
Thus, this PR effectively raises the skill ceiling of Medbay while keeping its skill floor at the same level.

**Comment by ATH1909** [comment]
> ITS NOT LIKE REPLICA PODS EXIST JUST SAYING CLONE REMOVAL HATERS

Replica pods are probably gonna get tweaked too if cloning gets removed; please actually read the OP of the cloning removal PR.

**Comment by ATH1909** [comment]
> This fails to actually solve the issue. Cloning devalues life as a whole, the problem is how people act before they're cloned because of it, not how they act after they're cloned.

Ah yes, because we all fling ourselves into danger recklessly, safe in the knowledge that the chimps in medbay will _definitely_ notice that we're dead and _definitely_ take the time to find our corpses and clone us, and _none_ of us care about the 10+ minute timeout (which is close to the maximum non-perma imprisonment length that sec can assign, by the way) that dying by flinging oneself into danger can cause.

What I'm trying to say is this: I (and other players as well, I assume) _already_ have absolutely no faith in the competency of Medbay or in the ability of medical doctors to operate even the "one click magic revive machine", so I kind of doubt that incresing the punishment for death is going to make us fear death even _more_.

But let's assume that this _does_ impact how we play. Let's say that this change makes everyone super afraid of death and its consequences. Why is this a good thing? Often, the things that we voluntary risk death to do are _cool_ things, like actually standing up to/trying to stop that murderboner. If you succeed, great, you have a cool story to tell. If you fail, oh well, losing is part of the game, but hey, you might get rezzed later.

Also, consider that, generally, the fear of death causes people to prepare for situations that can often cause it. What sorts of preparations do they make? Snatching flashes, making cablecuffs, grabbing a full set of tools, grabbing insuls, tiding for eye protection... Hm, what does that sound a lot like? If you guess "tiding and powergaming", you'd be correct, because preparing for deadly situations is half of the damn point of (/reason to attempt) tiding and/or powergaming.

**Comment by HomocidlePancakes** [comment]
Wew. Le 5 stages of grief in action. We went through denial, anger and are now bargaining.

But seriously, this will get insta closed once the 24 hours pass. It’s a kneejerk, so it will be given the finger by the coders. It will create so many fucking issues for the admins, so they won’t approve of this either. Not to
mention it implies a change to rules meaning it needed a headmin approval before being opened.

**Comment by Shadowflame909** [comment]
Any change to cloning would require changing the rules HomocidlePancakes.

**Comment by HomocidlePancakes** [comment]
I guess, since escalation exists. But this PR straight up adds a new rule while the removal PR just changes the rule dynamic, w/o necessarily changing the rules.

**Comment by OverlyGeneric** [comment]
There isn't a problem with people remembering their death. 

Cloning needs to be changed to the last resort than a go-to.

I've suggested recycling the experimental cloner's code and add other risks that exponentially grow as someone is cloned.

The risks could start like a genetic defect, brain trauma, conscious swapping, lack of consciousness, and finally spewing out an abomination.

The abomination isn't deadly but maybe similar to the meatcube from Goonstation.

**Comment by Couls** [comment]
I think you misunderstood the intention of the removal PR

**Comment by blobbernaut** [comment]
"Why use the other, more creative (for now; just wait until someone posts a step by step process for how to do the headswapping technique in the medical doctor guide on the wiki)/interactive means of reviving someone when the magic one button revive machine exists?"

I want to ask you if you have thought about this statement in any depth, or just shot it off.

What makes those other methods like headswapping interesting? Are they fundamentally more interactive? More enjoyable? 

No, they're just interesting because they're new. Soon they'd grow old, and then they'd become a chore, and then somebody will make a pull request for some science machine that will do it automatically because it's mind-numbingly boring doing it for the hundredth time. Not to mention, it's just more ridiculously magical to swap a person's braindead head than to clone a copy.

This isn't so bad with things you have to do maybe once a round, like research, setting up the engine, getting crates to sell, but if you're relying on something like that multiple times a shift, it's just painful. 


Thing is that they aren't ACTUALLY more interesting. They're just novel. Don't confuse them.


Cloning has been a reliable, interesting and fundamental part of the game for a solid decade, you should investigate the cause of why you suddenly seem to think it's so broken when clearly it isn't.

**Comment by Couls** [comment]
> No, they're just interesting because they're new. Soon they'd grow old, and then they'd become a chore, and then somebody will make a pull request for some science machine that will do it automatically because it's mind-numbingly boring doing it for the hundredth time. Not to mention, it's just more ridiculously magical to swap a person's braindead head than to clone a copy. 

if you think that doing the things a medical doctor is expected to do in a normal round is boring and you enjoy loading something into a machine and then clicking buttons then maybe playing medical just isn't for you. I'd suggest genetics.

**Comment by blobbernaut** [comment]
Yes, I can see it now, standard medical procedure in the 25th century when you find a dead body: 
Cut it off and swap it with a braindead husk's in order to defibrillate it. 

That's not medicine, that's necromancy. Your criticism is irrelevant.

**Comment by Couls** [comment]
What is the most interesting part of being a medical doctor? and quite a few other jobs have a repetitive process that is to be repeated throughout the shift. Xenobiology you just feed monkeys to slimes and farm extracts, genetics you repeat the same minigame with a few little modifiers until you get mutations. placing wiring for solars, ordering and dragging crates. placing food in the microwave and then serving it on a desk

**Comment by itseasytosee** [comment]
I don't see how this fixes the issue

**Comment by optimumtact** [comment]
Please do not making cloning Pr's without consulting me or ninja first.

It's bad enough watching one PR

**Comment by optimumtact** [comment]
Leave feedback here: https://tgstation13.org/phpBB/viewtopic.php?f=10&t=25214

**Comment by ShizCalev** [comment]
New life rule

**Comment by optimumtact** [comment]
yeah this doesn't really achieve anything
